vochain/transaction.go

```diff
@@ -332,40 +332,40 @@ func (app *BaseApplication) AddTx(vtx *VochainTx, commit bool) (*AddTxResponse,
 // VoteEnvelopeCheck is an abstraction of ABCI checkTx for submitting a vote
 // All hexadecimal strings should be already sanitized (without 0x)
 func (app *BaseApplication) VoteEnvelopeCheck(ve *models.VoteEnvelope, txBytes, signature []byte,
-	txID [32]byte, forCommit bool) (*models.Vote, error) {
-
+	txID [32]byte, forCommit bool) (*models.Vote, types.VoterID, error) {
 	// Perform basic/general checks
 	if ve == nil {
-		return nil, fmt.Errorf("vote envelope is nil")
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("vote envelope is nil")
 	}
 	process, err := app.State.Process(ve.ProcessId, false)
 	if err != nil {
-		return nil, fmt.Errorf("cannot fetch processId: %w", err)
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("cannot fetch processId: %w", err)
 	}
 	if process == nil || process.EnvelopeType == nil || process.Mode == nil {
-		return nil, fmt.Errorf("process %x malformed", ve.ProcessId)
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("process %x malformed", ve.ProcessId)
 	}
 	height := app.State.CurrentHeight()
 	endBlock := process.StartBlock + process.BlockCount
 
 	if height < process.StartBlock {
-		return nil, fmt.Errorf("process %x starts at height %d, current height is %d", ve.ProcessId, process.StartBlock, height)
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("process %x starts at height %d, current height is %d", ve.ProcessId, process.StartBlock, height)
 	} else if height > endBlock {
-		return nil, fmt.Errorf("process %x finished at height %d, current height is %d", ve.ProcessId, endBlock, height)
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("process %x finished at height %d, current height is %d", ve.ProcessId, endBlock, height)
 	}
 
 	if process.Status != models.ProcessStatus_READY {
-		return nil, fmt.Errorf("process %x not in READY state - current state: %s", ve.ProcessId, process.Status.String())
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("process %x not in READY state - current state: %s", ve.ProcessId, process.Status.String())
 	}
 
 	// Check in case of keys required, they have been sent by some keykeeper
 	if process.EnvelopeType.EncryptedVotes &&
 		process.KeyIndex != nil &&
 		*process.KeyIndex < 1 {
-		return nil, fmt.Errorf("no keys available, voting is not possible")
+		return nil, types.VoterID{}.Nil(), fmt.Errorf("no keys available, voting is not possible")
```

Comment:
since voterID is a return value, you might create the voterID var in the begging and return `voterID.Nil()` instead of `types.VoterID{}.Nil()`. I think that would be cleaner.